api/v1alpha1/serverless_helpers.go

```diff
@@ -16,6 +16,15 @@ func (s *Serverless) IsCondition(conditionType ConditionType) bool {
 	) != nil
 }
 
+func (s *Serverless) GetCondition(conditionType ConditionType) (metav1.Condition, bool) {
```

Comment:
I think in this case you don't need to create your own function because the metav1 package already provides such functionality to find the right condition ( read more [here](https://pkg.go.dev/k8s.io/apimachinery@v0.25.4/pkg/api/meta) ). [Here](https://github.com/kyma-project/serverless-manager/blob/main/api/v1alpha1/serverless_helpers.go#L14-L16) is an example in our code how to use it